larray/viewer.py

```diff
@@ -2626,11 +2777,11 @@ def edit(obj=None, title='', minvalue=None, maxvalue=None, readonly=False, depth
 
     dlg = MappingEditor(parent) if hasattr(obj, 'keys') else ArrayEditor(parent)
     if dlg.setup_and_check(obj, title=title, minvalue=minvalue, maxvalue=maxvalue, readonly=readonly):
-        if parent:
-            dlg.show()
-        else:
-            dlg.exec_()
+        if isinstance(obj, str):
+            dlg.updateRecentFileActions(obj)
```

Comment:
This code will never be called (because if obj WAS str, it should be converted to a Session by now).